Agrega Bootstrap y pruebas en Cypress a Clase 8

gruvix opened the pull request:
estilos bonitos
bugfixes
mejoras de sintaxis
pruebas en cypress

gruvix committed:
add bootstrap

gruvix committed:
add reference to bootstrap in html's

gruvix committed:
add install instructions

gruvix committed:
Renombre funciones

renombre funciones para mejorar la comprension futura

gruvix committed:
refactor click "Siguiente"

reformulado del resultado de clickear siguiente para que sea mas facil distinguir las partes del codigo
Ahora se fija si hay un error proveniente de validarCantidadFamiliares
si no hay error funciona normalmente

gruvix committed:
Revert "Renombre funciones"

This reverts commit 145e6d44d7641a52635a52cfc319e7f2faf62ebf.

gruvix committed:
removi retorno predeterminado de validacion de cantidad

gruvix committed:
renombre validar input  a validarCantidadFamiliares

para que sea mas explícito con lo que hace

gruvix committed:
add bootstrap

agrega los estilos de bootstrap a la mayoria del html, mejorando visualmente el contenido

gruvix committed:
cambia la clase de los inputs de edades

la nueva clase proviene de bootstrap, dejando un estilo mas bonito
la clase anterior se borra de css ya no se usa

gruvix committed:
fix reiniciar nuevamente remueve personas

gruvix committed:
refactor llamado a validarEdad

ahora se define la constante edad antes de validarla en otra funcion

gruvix committed:
add funcion ocultar bloque de errores de edad

gruvix committed:
error cantidad ahora tiene su propio bloque

ya no se muestra adentro del bloque de seleccion de cantidad, sino que aparece como un error aparte

gruvix committed:
add ocultar bloque de inputs de las edades

ahora el bloque de los inputs se oculta y se muestra cuando es necesario

gruvix committed:
fix instrucciones de instalacion erroneas

gruvix committed:
mejora de sintaxis

cambia el nonmbre de varias funciones y variables para mejor comprension del script
no se modifica la funcionalidad

gruvix committed:
renombra funcion validarEdad

renombra la funcion validarEdades por un titulo mas correcto a lo que cumple la funcion

gruvix committed:
renombra funcion a plural

la funcion manejarErroresEdades maneja múltiples errores

gruvix committed:
change magic numbers validaciones

ahora se usan constantes para las validaciones de la cantidad y las edades

gruvix committed:
renombra funcion de reiniciado

gruvix committed:
fix ya no se muestran errores viejos junto a los nuevos

los errores de edades ya no permanecen junto a nuevos errores, primero se limpian y luego se vuelven a crear

gruvix committed:
improve manejarErroresEdades

se inviertió la funcionalidad de manejarErroresEdades

gruvix committed:
ocultar resultados si hay errores

se ocultan los resultados viejos de maximo, minimo y promedio si hay errores de ingreso

gruvix committed:
limpieza de espacios innecesarios

gruvix committed:
mejora de sintaxis de manejarErroresEdades

ahora se declara una constante antes de comparar manejarErroresEdades
se renombra la variable hayErrores a simplemente "error"
se renombra gente a inputsEdades, ya que explica mejor lo que lleva la variable

gruvix committed:
add json

gruvix committed:
npm init

gruvix committed:
add node modules script quick execs

gruvix committed:
add cypres config files

gruvix committed:
test visit page

gruvix committed:
fix gente no existe

gruvix committed:
fix const y variable de error se llamaban igual

gruvix committed:
fix errorEdad mal nombrado

gruvix committed:
add prueba de cantidad 0

gruvix committed:
refactor prueba cantidad 0

+

gruvix committed:
add prueba easter egg

gruvix committed:
add prueba de valores esperados

agrega edades y evalua los valores esperados de edad mayor, edad menor y edad promedio

gruvix committed:
add prueba error maximo cantidad

gruvix committed:
add prueba edad vacia

gruvix committed:
cambia menor y mayor edad en la prueba de lugar

gruvix committed:
add prueba error edad con decimal

gruvix committed:
change nombre de pruebas de tarea 1

gruvix committed:
Update package-lock.json

gruvix committed:
agrega titulos a los html

GonzaloAlt commented on cypress/e2e/test.cy.js:
```diff
@@ -0,0 +1,58 @@
+describe('Pruebas de tarea 1', () => {
+  const TAREA1 = 'http://localhost:8000/tarea1'
+
+  it('agrega 0 en la cantidad de personas y espera un error', () => {
```
Esto tal vez podría ser `'debería arrojar error si se ingresa una cantidad inválida de personas'`, si vos mañana decidís que la cantidad valida es mayor a 5 por ejemplo, solo cambias el valor de la constante.

GonzaloAlt commented on cypress/e2e/test.cy.js:
```diff
@@ -0,0 +1,58 @@
+  const TAREA1 = 'http://localhost:8000/tarea1'
+
+  it('agrega 0 en la cantidad de personas y espera un error', () => {
+    const CANTIDAD = 0;
```
`const CANTIDAD_INVALIDA = 0;` tal vez podría ser mas descriptivo

GonzaloAlt commented on cypress/e2e/test.cy.js:
```diff
@@ -0,0 +1,58 @@
+    cy.get('#error-cantidad').should('have.text', 'el valor debe ser igual o mayor a 1');
+  })
+
+  it('agrega 101 en la cantidad de personas y espera un error', () => {
```
`'debería arrojar error si se ingresa una cantidad superior a la permitida'` por ejemplo.

GonzaloAlt commented on cypress/e2e/test.cy.js:
```diff
@@ -0,0 +1,58 @@
+  })
+
+  it('agrega 101 en la cantidad de personas y espera un error', () => {
+    const CANTIDAD = 101;
```
`const CANTIDAD_INVALIDA = 0;`

GonzaloAlt commented on cypress/e2e/test.cy.js:
```diff
@@ -0,0 +1,58 @@
+    cy.get('#error-cantidad').should('have.text', 'el valor debe ser menor de 100');
+  })
+
+  it(`agrega 1 en la cantidad de personas y chekea el easter egg`, () => {
```
`'debería mostrar el easter egg si...'`

GonzaloAlt commented on cypress/e2e/test.cy.js:
```diff
@@ -0,0 +1,58 @@
+    cy.get('#easter-egg').should('be.visible');
+  })
+
+  it(`deja una edad con un decimal y espera un error`, () => {
```
`'debería mostrar un mensaje de error si se ingresa una edad con decimal'`

GonzaloAlt commented on cypress/e2e/test.cy.js:
```diff
@@ -0,0 +1,58 @@
+  })
+
+  it(`deja una edad con un decimal y espera un error`, () => {
+    const CANTIDAD = 1;
```
`const CANTIDAD_PERSONAS = 1;`

GonzaloAlt commented on cypress/e2e/test.cy.js:
```diff
@@ -0,0 +1,58 @@
+    cy.get('#campos-decimales').should('be.visible');
+  })
+
+  it(`deja una edad vacía y espera un error`, () => {
```
`'debería mostrar un error al realizar los cálculos si el campo edad es vacío'`

gruvix committed:
Actualizó los títulos de los <it>s en las pruebas de Cypress

Se cambiaron los títulos de los IT en las pruebas de Cypress para que sean más descriptivos.

gruvix committed:
agrega beforeEach para reducir codigo en todas las pruebas

ya no hace falta especificar cy.visit en cada prueba, un beforeEach hace la visita a la página en cada una

gruvix committed:
mejora nombre de constantes en pruebas de cypress

gruvix committed:
actualiza nombre de prueba de resultados de tarea 1
